src/libsystemd-network/lldp-neighbor.c

```diff
@@ -292,10 +293,22 @@ int lldp_neighbor_parse(sd_lldp_neighbor *n) {
 
                         break;
 
-                case SD_LLDP_TYPE_PRIVATE:
+                case SD_LLDP_TYPE_PRIVATE: {
+                        uint8_t subtype = SD_LLDP_OUI_SUBTYPE_MUD_USAGE_DESCRIPTION;
+
                         if (length < 4)
                                 log_lldp("Found private TLV that is too short, ignoring.");
 
+                        /* RFC 8520: MUD URL */
+                        if (memcmp(p, SD_LLDP_OUI_MUD, sizeof(SD_LLDP_OUI_MUD)) == 0 &&
+                            memcmp(p + sizeof(SD_LLDP_OUI_MUD), &subtype, 1) == 0) {
```

Comment:
it's a single byte only, hence don't bother with memcmp(), just do `p[sizeof(SD_LLDP_OUI_MUD)] == SD_LLDP_OUI_SUBTYPE_MUD_USAGE_DESCRIPTION` here.

Reply:
nice thanks